src/engine/sequencer.js

```diff
@@ -54,28 +49,18 @@ Sequencer.prototype.stepThreads = function (threads) {
             var activeThread = threads[i];
             if (activeThread.status === Thread.STATUS_RUNNING) {
                 // Normal-mode thread: step.
-                this.stepThread(activeThread);
+                this.startThread(activeThread);
             } else if (activeThread.status === Thread.STATUS_YIELD) {
                 // Yield-mode thread: check if the time has passed.
-                YieldTimers.resolve(activeThread.yieldTimerId);
-                numYieldingThreads++;
+                if (!YieldTimers.resolve(activeThread.yieldTimerId)) {
+                    numYieldingThreads++;
+                }
```

Comment:
That's a pretty good catch I think.

I think we should reset `numYieldingThreads` before the `for` loop, i.e., it should reset every time we are going to iterate across all the threads.

Also...I still think we need to go over this whole yielding flow and how things are treated generally, especially if we intend to implement yielding reporters...